tests/test_base_quantity.py

```diff
@@ -95,5 +96,8 @@ def test_custom_serialization():
 
     assert model_dump["voltage"] == str(Voltage(10.0, "volt"))
 
+    model_dump = component.model_dump(context={"magnitude_only": True})
+    assert model_dump["voltage"] == 10.0
+
     model_dump = component.model_dump(mode="json", context={"magnitude_only": True})
-    assert model_dump["voltage"] == str(10.0)
+    assert model_dump["voltage"] == "10.0"
```

Comment:
I feel skeptical about this. I think line 100 is correct - this value should be a number. Why is the JSON case different?

Reply:
It returns a string on purpose. I added this behavior so that pydantic knows how to serialize the pint.Quantities.

Reply:
Understood. I'm agreeing that serialization of magnitude + units should be a string, but then suggesting that magnitude only shouldn't change type, just like line 100 when serializing to a dict.

Reply:
Got it. I can add that into the serialization logic.

Reply:
@daniel-thom This should be fixed now.